x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -989,10 +990,12 @@ private AbstractRunnable fillGapRunnable(SparseFileTracker.Gap gap, RangeMissing
                     start,
                     start,
                     Math.toIntExact(gap.end() - start),
-                    progress -> gap.onProgress(start + progress)
+                    progress -> gap.onProgress(start + progress),
+                    ActionListener.wrap(unused -> {
+                        writeCount.increment();
+                        gap.onCompletion();
+                    }, gap::onFailure)
```

Comment:
Should this have been `failGapAndListener(gap, l, e)` (like in the enclosing ActionRunnable) instead of `gap::onFailure`?

Reply:
Thank you! Yes, we definitely need to fail the listener similar to the case if `fillCacheRange` implementation would throw an exception.